src/engraving/dom/edit.cpp

```diff
@@ -4596,18 +4596,21 @@ void Score::cmdTimeDelete()
         startSegment = selection().startSegment();
         endSegment   = selection().endSegment();
     }
-
     if (!isMaster() && masterScore()) {
-        Measure* masterStartMeas = masterScore()->tick2measure(startSegment->tick());
-        Measure* masterEndMeas = masterScore()->tick2measure(endSegment->tick());
-        if (endSegment->isEndBarLineType()) {
-            Measure* prevEndMeasure = masterEndMeas->prevMeasure();
-            masterEndMeas = prevEndMeasure ? prevEndMeasure : masterEndMeas;
-        }
-        Segment* masterStartSeg
-            = masterStartMeas ? masterStartMeas->findSegment(startSegment->segmentType(), startSegment->tick()) : startSegment;
-        Segment* masterEndSeg = masterEndMeas ? masterEndMeas->findSegment(endSegment->segmentType(), endSegment->tick()) : endSegment;
-        masterScore()->doTimeDelete(masterStartSeg, masterEndSeg);
+        if (!endSegment) {
+            masterScore()->doTimeDelete(startSegment, endSegment);
```

Comment:
We do need to use a `masterStartSegment` here rather than `startSegment`

Reply:
Would have been an easy change, but @miiizen's seems better